### ISSUE TITLE
Markdown plugin breaks namepaths of objects with special characters in the name

### ISSUE THREAD

**Issue by Oblongmana** [jsdoc3/jsdoc#1035]
The Namepaths page of the jsdoc docs indicates the use of quotes for handling special characters in namepaths, providing this example (see http://usejsdoc.org/about-namepaths.html):
```
/** @namespace */
var chat = {
    /**
     * Refer to this by {@link chat."#channel"}.
     * @namespace
     */
    "#channel": {
        /**
         * Refer to this by {@link chat."#channel".open}.
         * @type {boolean}
         * @defaultvalue
         */
        open: true,
        /**
         * Internal quotes have to be escaped by backslash. This is
         * {@link chat."#channel"."say-\"hello\""}.
         */
        'say-"hello"': function (msg) {}
    }
};
```

However, if you turn on the markdown plugin (using the `marked` parser - as is the only option in current version of jsdoc), this plugin breaks the tags with namepaths which include special characters by converting the character `"` into the html entity `&quot;`. To demonstrate the issue, save the code *above* as `chat.js`, save the conf file *below* as `jsdoc.conf.json`, and run 
`$ jsdoc -c jsdoc.conf.json chat.js && open out/chat.__channel_.html`
```
//conf file to demo the issue, save as jsdoc.conf.json
{
  "plugins": ["plugins/markdown"],
  "markdown": {
    "parser": "marked",
    "hardwrap": true
  }
}
```

Note that in the page that's opened (the doc page for `#channel`), the links shown in the source are not turned into links

To solve the issue (at least for myself), I essentially copied the `escapeUnderscores(...)` method in `jsdoc/util/markdown` - creating an `unencodeQuotes(...)` method, that grabs the `{@ ... }` tags, and converts any `&quot;` in the tags back into `"`. Will open a PR shortly with this change - hopefully someone with a bit more knowledge of the jsdoc project can provide some insight as to whether that's sufficient and won't break anything else, or needs some changing!

**Comment by hegemonic** [comment]
Thanks for flagging this issue and contributing a proposed fix. I'll need to put on my thinking cap to make sure the fix doesn't break anything. I'll try to do that soon.

**Comment by hegemonic** [comment]
This is fixed on master and releases/3.4. The fix will be included in JSDoc 3.4.1.
